src/chainstate/stacks/mod.rs

```diff
@@ -552,6 +584,16 @@ pub struct MultisigSpendingCondition {
     pub signatures_required: u16,
 }
 
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct OrderIndependentMultisigSpendingCondition {
```

Comment:
There is a lot of code duplication involved in implementing this structure and its serialization/deserialization. Consider factoring out the fields shared between the different multisig spending conditions, like this:
```rust
pub struct MultisigTransactionData {
    pub signer: Hash160,
    pub nonce: u64,  // nth authorization from this account
    pub tx_fee: u64, // microSTX/compute rate offered by this account
    pub fields: Vec<TransactionAuthField>,
    pub signatures_required: u16,
}

pub struct MultisigSpendingCondition {
    pub hash_mode: MultisigHashMode,
    pub data: MultisigTransactionData,
}

pub struct OrderIndependentMultisigSpendingCondition {
    pub hash_mode: OrderIndependenMultisigHashMode,
    pub data: MultisigTransactionData,
}
```

Reply:
I'm going to push back against this.  The existing authentication fields are **consensus-critical**, which means that we should be treating them as read-only code.  It's much safer to implement a new order-independent multisig spending condition from scratch than to try and refactor the existing code.  I'm afraid I'm going to have to insist on this as a condition for merging this PR.  Hope you both understand! :pray: